clang-tools-extra/clang-doc/Representation.h

```diff
@@ -331,6 +336,11 @@ struct Info {
   llvm::SmallString<128> Path;          // Path of directory where the clang-doc
                                         // generated file will be saved
 
+  // The name used for the file that this info is documented in.
+  // In the JSON generator, infos are documented in files with mangled names.
+  // Thus, we keep track of the physical filename for linking purposes.
+  SmallString<16> DocumentationFileName;
```

Comment:
Why 16? I'm guessing no mangled name will be only 16 chars (unless its C and not C++). For now this is fine, but we should try and revisit our string usage and try to make it consistent. Some day, I'd like to start interning strings as well, and then just dropping them w/ an arena, ala `UniqueStringSaver`.